container-tests/tests/src/test/java/org/hyperledger/besu/tests/container/helpers/ContractOperations.java

```diff
@@ -138,10 +139,16 @@ public static String sendLogEventAndReturnTransactionHash(
     return logReceiptResult.getTransactionHash();
   }
 
-  public static String generateHexString(final long number) {
-    final StringBuilder randomValue = new StringBuilder(Long.toHexString(number));
+  public static String generate64BytesHexString(final long number) {
+    final String str = Long.toHexString(number);
 
-    while (randomValue.length() < 64) {
+    return prependZeroesToHexString(str, 64);
+  }
+
+  @NotNull
+  public static String prependZeroesToHexString(final String hexString, final int lenRequested) {
+    final StringBuilder randomValue = new StringBuilder(hexString);
```

Comment:
there's no randomness here, is there? hexStringToPrependZeroes ?

Reply:
Done